src/viam/sdk/tests/test_robot.cpp

```diff
@@ -31,30 +31,36 @@ using namespace viam::sdk;
 
 BOOST_AUTO_TEST_SUITE(test_robot)
 
-template <typename Lambda>
-void server_to_client_pipeline(Lambda&& func) {
-    MockRobotService service;
-    auto manager = service.resource_manager();
-
-    manager->add(std::string("mock_generic"), generic::MockGeneric::get_mock_generic());
-    manager->add(std::string("mock_motor"), motor::MockMotor::get_mock_motor());
-    manager->add(std::string("mock_camera"), camera::MockCamera::get_mock_camera());
-
-    ::grpc::ServerBuilder builder;
-    builder.RegisterService(&service);
-
-    std::unique_ptr<grpc::Server> server = builder.BuildAndStart();
-
+// Basically a copy of the client_to_mock_pipeline test utility. This version
+// has multiple mock resources and gives the passed in test case access to both
+// the robot client and the mock robot service.
+template <typename F>
+void robot_client_to_mocks_pipeline(F&& test_case) {
```

Comment:
Couldn't quite replace this one with `client_to_mocks_pipeline`  as it was a subtly different setup; see my comment above.

Reply:
Is the issue just that we register multiple resources? We could make the `client_to_mocks_pipeline` take a vec of resources to register rather than a single resource and register everything in the vec... not sure this is actually better though, given the sort of unique space that the `robot` occupies and the slightly weird structure of having every other test file pass a vec with only a single element.

I think I'm convincing myself that it's fine as is, but including comment for posterity/to see if others have thoughts.

Reply:
I think it is fine how you have it.